deltachat-ios/Helper/Utils.swift

```diff
@@ -42,12 +53,31 @@ struct Utils {
         return window?.safeAreaInsets.bottom ?? 0
     }
 
+    public static func makeDeltaChatInvitationQRCode(from url: URL) -> String? {
+        guard url.isDeltaChatInvitation,
+              let components = URLComponents(url: url, resolvingAgainstBaseURL: true),
+              let fragment = components.fragment else { return nil }
+              
+        let fragments = fragment.split(separator: "&")
+
+        guard let fingerprint = fragments.first else { return nil }
+
+        var qrcodeString = "OPENPGP4FPR:"
+        qrcodeString.append(String(fingerprint))
+        qrcodeString.append("#")
+
+        let queryParams = fragments.dropFirst().joined(separator: "&")
+        qrcodeString.append(queryParams)
+
+        return qrcodeString
+    }
+
```

Comment:
this conversion is not needed - you can pass an i.delta.chat invite-link directly to handleQRCode() (as core already handles `OPENPGP4FPR:` and `https://i.delta.chat` equally; sorry for the additional work, i could have pointed that out clearer)

Reply:
That's good to know, thank you!